Src/main.c

```diff
@@ -102,16 +102,47 @@ int main(void)
 
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
+
+  // Start ADC
+  HAL_ADC_Start(&hadc); 
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
+
+  // Declaring variables
+  const int TIMER_PERIOD = 60000;
+  const int DESIRED_FREQUENCY = 50;
+  const int BASE_ON_TIME = 1;
+  uint16_t raw_value; // Use 16bit value since ADC is 12 bit (0-4095)
+  float offset;
+  float pulse;
```

Comment:
Good practice is to initialize values when you declare them. Consider setting these to 0

Reply:
Ahaha, reminds me of a certain Path Manager 1 in 5 issue LOL